internal/providers/vmware/vmware_amd64.go

```diff
@@ -77,6 +79,39 @@ func fetchRawConfig(f *resource.Fetcher) (config, error) {
 	}, nil
 }
 
+func fetchUrlConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
+	rawUrl, err := getVariable(f, "ignition.config.url")
+	if err != nil || rawUrl == "" {
+		rawUrl, err = getVariable(f, "coreos.config.url")
+	}
+
+	if err != nil || rawUrl == "" {
+		f.Logger.Info("no config URL provided")
+		return types.Config{}, report.Report{}, providers.ErrNoProvider
+	}
+
+	f.Logger.Debug("found url: %q", rawUrl)
+
+	url, err := url.Parse(rawUrl)
+	if err != nil {
+		f.Logger.Err("failed to parse url: %v", err)
+		return types.Config{}, report.Report{}, err
+	}
+
+	if url == nil {
+		return types.Config{}, report.Report{}, providers.ErrNoProvider
+	}
+
+	data, err := f.FetchToBuffer(*url, resource.FetchOptions{
+		Headers: resource.ConfigHeaders,
+	})
+	if err != nil {
+		return types.Config{}, report.Report{}, err
+	}
+
+	return util.ParseConfig(f.Logger, data)
```

Comment:
There's some code duplication here, since `ParseConfig` is being called for the other (non-url) path in the caller.

I think it would be better to move the data-decoding for the non-url case into `fetchRawConfig` and rename it to `fetchDataConfig`, and have `fetchUrlConfig` return the unparsed data. Then, both paths would go through `ParseConfig` at the end of `FetchConfig`, and each function (`fetchDataConfig` and `fetchUrlConfig`) would be doing equivalent work instead of fairly different work.

Reply:
Yes, makes sense, thanks for the hint!